### PR TITLE
Added Author Checklist to PR template.

### PR DESCRIPTION
**Pull request by markdorison**
## Description
Added author checklist items to PR template.

## Motivation / Context
Discussion w/ Adam & Alfonso.

## Testing Instructions / How This Has Been Tested
N/A

## Screenshots
N/A

## Documentation
N/A

### CODE REVIEW

**Comment by markdorison** [comment]
@adamzimmermann @agarzola Thoughts?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -15,3 +15,9 @@
 ## Documentation
 <!--- Do any of the changes affect the accuracy of existing documentation? --->
 <!--- Do they warrant new sections within the current documentation? --->
+
+## PR Author Checklist
+- [ ] My PR description includes a reference to a ticket or other "motivation" for these changes.
+- [ ] My PR description includes details of how I tested these changes.
+- [ ] My PR description includes testing instructions for reviewers.
```
Comment (agarzola) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] My PR description includes testing instructions for reviewers with links to the testing environment, if applicable.
```
Thoughts?

**Review by agarzola — COMMENTED** [review]
This looks good. I have one suggestion.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -15,3 +15,9 @@
...
+- [ ] My code includes documentation updates including updates to existing documentation or new sections if applicable.
```
Comment (adamzimmermann) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] My code includes documentation updates to existing documentation or new sections if applicable.
```

**Review by adamzimmermann — APPROVED** [review]
I like it.

**Comment by adamzimmermann** [comment]
Can we merge this?

**Pull request merged**